coi/src/utils.rs

```diff
@@ -111,6 +114,23 @@ pub(crate) mod serde_duration_as_days {
     }
 }
 
+/// # Panics
+///
+/// This method is primarily used for testing.
+/// It simply converts an Array into a normalized Array.
+///
+/// Will panic if the array cannot be normalized.
+pub fn normalize_array<const N: usize>(array: [f32; N]) -> [f32; N] {
+    let embedding: Embedding<Ix1> = Array::from_vec(array.into()).into();
+    embedding
+        .normalized()
+        .unwrap()
+        .as_slice()
+        .unwrap()
+        .try_into()
+        .unwrap()
```

Comment:
:art: we can create the embedding directly from the array, also using `to_vec` instead of `as_slice` makes this more robust in case of non-continguous memory
```suggestion
    Embedding::from(array)
        .normalized()
        .unwrap()
        .to_vec()
        .try_into()
        .unwrap()
```

Reply:
yup!